src/user/components/SignUpCard.vue

```diff
@@ -0,0 +1,198 @@
+            v-model="newAccount.email"
+            name="email"
+            type="email"
+            required
+            class="input"
+          />
+        </div>
+
+        <div class="form--element">
+          <label for="password" class="label">Password:</label>
+          <TextInput
+            id="password"
+            ref="pwRef"
+            v-model="newAccount.pw.password"
+            name="password"
+            type="password"
+            minlength="8"
+            required
+            class="input"
+            @change="validatePassword"
+          />
+        </div>
+
+        <div class="form--element">
+          <label for="passwordConfirm" class="label">Confirm Password:</label>
+          <TextInput
+            id="passwordConfirm"
+            v-model="newAccount.pw.confirm"
+            name="passwordConfirm"
+            type="password"
+            minlength="8"
+            required
+            class="input"
+            @change="validatePassword"
+          />
+        </div>
+
+        <AppButton variant="primary" type="submit" class="btn">
+          Create account
+        </AppButton>
+
+        <AppButton
+          variant="secondary"
+          type="button"
+          class="btn"
+          @click="logInViaGoogle()"
+        >
+          <GoogleLogo variant="primary" class="btn-logo" />
+          Login with Google
+        </AppButton>
+      </form>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, ref, computed } from "vue";
+import AppButton from "@/common/AppButton.vue";
+import TextInput from "@/common/TextInput.vue";
+import GoogleLogo from "@/common/app-icon/GoogleLogo.vue";
+import { useRouter, useRoute } from "vue-router";
+import client from "../client";
+
+export default defineComponent({
+  name: "SignUpCard",
+  components: { AppButton, TextInput, GoogleLogo },
+
+  setup() {
+    const router = useRouter();
+    const route = useRoute();
+
+    const newAccount = ref({
+      name: "",
+      email: "",
+      pw: {
+        password: "",
+        confirm: ""
+      }
+    });
+
+    // const pwRef = ref<InstanceType<typeof TextInput>>();
+    const pwRef = ref();
+
+    const isPasswordMatch = computed(
+      () => newAccount.value.pw.password === newAccount.value.pw.confirm
+    );
+
+    const redirectTo = computed(() => {
+      const redirectTo = route.query.redirectTo as string;
+      return redirectTo || "/";
+    });
+
+    const validatePassword = () => {
+      if (isPasswordMatch.value) {
+        pwRef.value.textinput.input.setCustomValidity(""); // Reset HTML validation
+      } else {
+        pwRef.value.textinput.input.setCustomValidity("Passwords don't match");
+      }
+    };
+
+    const createAccount = () => {
+      validatePassword();
+      pwRef.value.textinput.input.reportValidity();
+      client.createUser(newAccount.value.email, newAccount.value.pw.password);
+      // .then(() => router.push(redirectTo.value));
```

Comment:
Is this not necessary so users are redirected as well? If it's not needed, let's delete the line instead of commenting it out.

Reply:
I left it there because that's what I wanted to do but I was getting a `Property 'then' does not exist on type 'void'.`

It looks like the implementation in `client.ts` doesn't return a promise. I've removed it for now, but something would need to go there or in `client.ts` to redirect the user. It might work in `client.ts` but thoughts on should it be there?